src/pages/test-report/components/vehicle-check/vehicle-check.ts

```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
       select(getVehicleChecks),
     );
 
-    vehicleChecks$.subscribe((vehicleChecks: VehicleChecks) => {
-      this.tellMeQuestionFault = vehicleChecks.tellMeQuestion.outcome;
-    });
+    const isSeriousMode$ = this.store$.pipe(
+      select(getTestReportState),
+      select(isSeriousMode),
+    );
+
+    const merged$ = merge(
+      vehicleChecks$.pipe(map((vehicleChecks: VehicleChecks) => {
+        this.tellMeQuestionFault = vehicleChecks.tellMeQuestion.outcome;
+        this.showMeQuestionFault = vehicleChecks.showMeQuestion.outcome;
+      })),
+      isSeriousMode$.pipe(map(toggle => this.isSeriousMode = toggle)),
+    );
 
+    this.subscription = merged$.subscribe();
   }
 
   ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
   }
 
   onTap = () => {
-    console.log('competency tapped');
+    this.addOrRemoveFault();
   }
 
   onPress = () => {
-    console.log('competency pressed');
+    this.addOrRemoveFault(true);
   }
 
   canButtonRipple = () => {
+    // TODO: Implement this
     return true;
   }
 
+  addOrRemoveFault = (wasPress: boolean = false): void => {
+    if (this.isRemoveFaultMode) {
+      this.removeFault();
+      return;
+    }
+
+    this.addFault(wasPress);
+  }
+
+  removeFault = (): void => {
+    // TODO: Implement this
+  }
+
+  addFault = (wasPress: boolean): void => {
+    if (this.isSeriousMode) {
+      this.store$.dispatch(new ShowMeQuestionSeriousFault());
+      this.store$.dispatch(new ToggleSeriousFaultMode());
+    }
+
+    if (wasPress) {
+      // TODO: Implement this
+    }
+  }
+
   hasDrivingFault = (): number => {
-    const hasDrivingFault = this.tellMeQuestionFault === CompetencyOutcome.DF;
-    return hasDrivingFault ? 1 : 0;
+    if (this.hasSeriousFault()) {
+      return 0;
+    }
+
+    if (this.hasDangerousFault()) {
+      return 0;
+    }
+
+    if (this.tellMeQuestionFault !== CompetencyOutcome.DF) {
+      return 0;
+    }
+
+    // if show me question is not DF then return 0
```

Comment:
comment placement is wrong, should be above the code it comments.

Reply:
It comments a code that does not yet exist. That's why it stands individually.